cfme/physical/physical_storage.py

```diff
@@ -0,0 +1,231 @@
+        return view.entities.properties.get_text_of("Description")
+
+    def _wait_for_state_change(self, desired_state, target, provider, view, timeout=300, delay=10):
+        """Wait for the Physical Storage to reach the desired state. This function waits an amount
+           of time due to wait_for.
+
+        Args:
+            desired_state (str): 'on' or 'off'
+            target (str): The name of the method that most be used to compare with the desired_state
+            provider (object): 'LenovoProvider'
+            view (object): The view that most be refreshed to verify if the value was changed
+            timeout (int): Specify amount of time (in seconds) to wait until TimedOutError is raised
+            delay (int): Specify amount of time (in seconds) to repeat each time.
+        """
+
+        def _is_state_changed():
+            self.refresh(provider, handle_alert=True)
+            return desired_state == getattr(self, target)()
+
+        wait_for(_is_state_changed, fail_func=view.browser.refresh, num_sec=timeout, delay=delay)
+
+    @property
+    def exists(self):
+        """Checks if the physical_Storage exists in the UI.
+
+        Returns: :py:class:`bool`
+        """
+        view = navigate_to(self.parent, "All")
+        try:
+            view.entities.get_entity(name=self.name, surf_pages=True)
+        except ItemNotFound:
+            return False
+        else:
+            return True
+
+    @cached_property
+    def db_id(self):
+        return self.appliance.physical_storage_id(self.name)
+
+    def wait_to_appear(self):
+        """Waits for the storage to appear in the UI."""
+        view = navigate_to(self.parent, "All")
+        logger.info("Waiting for the storage to appear...")
+        wait_for(
+            lambda: self.exists,
+            message="Wait for the storage to appear",
+            num_sec=1000,
+            fail_func=view.browser.refresh
+        )
+
+    def wait_for_delete(self):
+        """Waits for the storage to remove from the UI."""
+        view = navigate_to(self.parent, "All")
+        logger.info("Waiting for the storage to delete...")
+        wait_for(
+            lambda: not self.exists,
+            message="Wait for the storage to disappear",
+            num_sec=500,
+            fail_func=view.browser.refresh
+        )
+
+
+@attr.s
+class PhysicalStorageCollection(BaseCollection):
+    """Collection object for the :py:class:`cfme.infrastructure.host.PhysicalStorage`."""
+
+    ENTITY = PhysicalStorage
+
+    def select_entity_rows(self, *physical_storages):
+        """ Select all physical storage objects """
+        physical_storages = list(physical_storages)
+        checked_physical_storages = list()
+        view = navigate_to(self, 'All')
+
+        for physical_storage in physical_storages:
+            view.entities.get_entity(name=physical_storage.name, surf_pages=True).check()
+            checked_physical_storages.append(physical_storage)
```

Comment:
Required:
No need to append those as you are returning View.

Question:
why you are returning `view`?